pkg/deploy/gateway/gateway.go

```diff
@@ -252,11 +252,18 @@ func GetGatewayRouteConfig(deployContext *deploy.DeployContext, component string
 	pathRewrite := pathPrefix != "/" && stripPrefix
```

Comment:
The interaction between this and the below `if` on the `component` makes for some fun non-local entanglement of the callers and this method. Is this the quantum computing everyone is talking about?

Reply:
:D this is really just a draft to test if this approach works